com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/client/DatabaseQueryClient.java

```diff
@@ -0,0 +1,165 @@
+                resultSet = preparedStatement.executeQuery();
+                return buildDatabaseTableData(databaseTable, resultSet);
+            } finally {
+                JdbcUtil.closeQuietly(resultSet);
+                JdbcUtil.closeQuietly(preparedStatement);
+            }
+        });
+    }
+
+    private DatabaseTableData buildDatabaseTableData(String databaseTable, ResultSet resultSet) throws SQLException {
+        List<DatabaseTableColumnMetadata> databaseTableColumnsMetadata = parseDatabaseColumnsMetadata(resultSet.getMetaData());
+        List<DatabaseTableRowData> databaseTableRowsData = parseDatabaseRowsData(resultSet, databaseTableColumnsMetadata);
+        return ImmutableDatabaseTableData.builder()
+                                         .tableName(databaseTable)
+                                         .addAllTableColumnsMetadata(databaseTableColumnsMetadata)
+                                         .addAllTableRowsData(databaseTableRowsData)
+                                         .build();
+    }
+
+    private List<DatabaseTableColumnMetadata> parseDatabaseColumnsMetadata(ResultSetMetaData resultSetMetadata) throws SQLException {
+        List<DatabaseTableColumnMetadata> databaseColumnsMetadata = new ArrayList<>();
+        for (int currentColumnIndex = 1; currentColumnIndex <= resultSetMetadata.getColumnCount(); currentColumnIndex++) {
+            databaseColumnsMetadata.add(ImmutableDatabaseTableColumnMetadata.builder()
+                                                                            .columnName(resultSetMetadata.getColumnName(currentColumnIndex))
+                                                                            .columnType(resultSetMetadata.getColumnTypeName(currentColumnIndex))
+                                                                            .build());
+        }
+        return databaseColumnsMetadata;
+    }
+
+    private List<DatabaseTableRowData> parseDatabaseRowsData(ResultSet resultSet, List<DatabaseTableColumnMetadata> databaseColumnsMetadata)
+        throws SQLException {
+        List<DatabaseTableRowData> databaseTableRowsData = new ArrayList<>();
+        while (resultSet.next()) {
+            databaseTableRowsData.add(ImmutableDatabaseTableRowData.builder()
+                                                                   .putAllValues(collectRowValues(resultSet, databaseColumnsMetadata))
+                                                                   .build());
+        }
+        return databaseTableRowsData;
+    }
+
+    private Map<DatabaseTableColumnMetadata, Object>
+            collectRowValues(ResultSet resultSet, List<DatabaseTableColumnMetadata> databaseColumnsMetadata) throws SQLException {
+        Map<DatabaseTableColumnMetadata, Object> databaseTableRowValues = new HashMap<>();
+        for (int currentColumnIndex = 0; currentColumnIndex < databaseColumnsMetadata.size(); currentColumnIndex++) {
```

Comment:
The index starts from 0 here, but `ResultSet` counts is columns from 1. See: https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#getObject(int,%20java.util.Map)